src/Microsoft.OData.Core/MultipartMixed/ODataMultipartMixedBatchReaderStream.cs

```diff
@@ -0,0 +1,684 @@
+        /// to be more strict.
+        /// </remarks>
+        private ODataBatchOperationHeaders ValidatePartHeaders(ODataBatchOperationHeaders headers, out bool isChangeSetPart)
+        {
+            string contentType;
+            if (!headers.TryGetValue(ODataConstants.ContentTypeHeader, out contentType))
+            {
+                throw new ODataException(Strings.ODataBatchReaderStream_MissingContentTypeHeader);
+            }
+
+            if (MediaTypeUtils.MediaTypeAndSubtypeAreEqual(contentType, MimeConstants.MimeApplicationHttp))
+            {
+                isChangeSetPart = false;
+
+                // An operation part is required to have application/http content type and
+                // binary content transfer encoding.
+                string transferEncoding;
+                if (!headers.TryGetValue(ODataConstants.ContentTransferEncoding, out transferEncoding) ||
+                    string.Compare(transferEncoding, ODataConstants.BatchContentTransferEncoding, StringComparison.OrdinalIgnoreCase) != 0)
+                {
+                    throw new ODataException(Strings.ODataBatchReaderStream_MissingOrInvalidContentEncodingHeader(
+                        ODataConstants.ContentTransferEncoding,
+                        ODataConstants.BatchContentTransferEncoding));
+                }
+            }
+            else if (MediaTypeUtils.MediaTypeStartsWithTypeAndSubtype(contentType, MimeConstants.MimeMultipartMixed))
+            {
+                isChangeSetPart = true;
+
+                if (this.changesetBoundary != null)
+                {
+                    // Nested changesets are not supported
+                    throw new ODataException(Strings.ODataBatchReaderStream_NestedChangesetsAreNotSupported);
+                }
+            }
+            else
+            {
+                throw new ODataException(Strings.ODataBatchReaderStream_InvalidContentTypeSpecified(
+                    ODataConstants.ContentTypeHeader,
+                    contentType,
+                    MimeConstants.MimeMultipartMixed,
+                    MimeConstants.MimeApplicationHttp));
+            }
+
+            return headers;
+        }
+
+        /// <summary>
+        /// Parse the content type header value to retrieve the boundary and encoding of a changeset.
+        /// </summary>
+        /// <param name="contentType">The content type to parse.</param>
+        private void DetermineChangesetBoundaryAndEncoding(string contentType)
+        {
+            Debug.Assert(!string.IsNullOrEmpty(contentType), "Should have validated that non-null, non-empty content type header exists.");
+
+            ODataMediaType mediaType;
+            ODataPayloadKind readerPayloadKind;
+            MediaTypeUtils.GetFormatFromContentType(
+                contentType,
+                new ODataPayloadKind[] { ODataPayloadKind.Batch },
+                new ODataMediaTypeResolver(),

```

Comment:
>ODataMediaTypeResolver() [](start = 20, length = 24)

In 6.x it was ODataMediaTypeResolver.DefaultMediaTypeResolver. Need to revisit this